skimage/morphology/convex_hull.py

```diff
@@ -18,6 +18,58 @@ def _offsets_diamond(ndim):
     return offsets
 
 
+def _check_coords_in_hull(gridcoords, hull_equations, tolerance):
+    r"""Checks all the co-ordinates for inclusiveness in the convex hull.
+
+    Parameters
+    ----------
+    gridcoords : (M, N) ndarray
+        Co-ordinates of ``N`` points in ``M`` dimensions.
+    hull_equations : (M, N) ndarray
+        Hyperplane equations of the facets of the convex hull.
+    tolerance : float
+        Tolerance when determining whether a point is inside the hull. Due
+        to numerical floating point errors, a tolerance of 0 can result in
+        some points erroneously being classified as being outside the hull.
+
+    Returns
+    -------
+    coords_in_hull : ndarray of bool
+        Binary 1D ndarray representing points in n-dimensional space
+        with value ``True`` set for points inside the convex hull.
+
+    Notes
+    -----
+    Check inclusiveness of co-ordinates in convex hull requires intermediate
+    calculations of dot products which are memory-intensive. Thus, the convex
+    hull equations are checked individually with all co-ordinates to keep
+    within the memory limit.
+
+    References
+    ----------
+    .. [1] https://github.com/scikit-image/scikit-image/issues/5019
+
+    """
+    ndim, n_coords = gridcoords.shape
+    n_hull_equations = hull_equations.shape[0]
+    coords_in_hull = np.ones(n_coords, dtype=np.bool_)
+
+    # Pre-allocate arrays to cache intermediate results for reducing overheads
+    dot_array = np.zeros(n_coords, dtype=np.float64)
+    test_ineq_temp = np.zeros(n_coords, dtype=np.float64)
+    coords_single_ineq = np.zeros(n_coords, dtype=np.bool_)
```

Comment:
```suggestion
    dot_array = np.empty(n_coords, dtype=np.float64)
    test_ineq_temp = np.empty(n_coords, dtype=np.float64)
    coords_single_ineq = np.empty(n_coords, dtype=np.bool_)
```

empty can save time by not initializing to zeros